src/bom/build.gradle.kts

```
@@ -87,7 +87,13 @@ dependencies {
         apiv("javax.mail:mail")
         apiv("jcharts:jcharts")
         apiv("junit:junit", "junit4")
-        apiv("org.codehaus.groovy:groovy-all")
+        apiv("org.codehaus.groovy:groovy", "groovy")
+        apiv("org.codehaus.groovy:groovy-datetime", "groovy")
+        apiv("org.codehaus.groovy:groovy-jmx", "groovy")
+        apiv("org.codehaus.groovy:groovy-json", "groovy")
+        apiv("org.codehaus.groovy:groovy-jsr223", "groovy")
+        apiv("org.codehaus.groovy:groovy-sql", "groovy")
+        apiv("org.codehaus.groovy:groovy-templates", "groovy")
```

Can you please use `org.codehaus.groovy:groovy-bom` instead?

```suggestion
        api(platform("org.codehaus.groovy:groovy-bom:${"groovy".v}"))
```

---

That looks really nice, but when I try it out, it gives me the following error:
```
Build jmeter FAILURE reason:                                
    org.gradle.internal.exceptions.LocationAwareException: Build file '/home/felix/Developer/fschumacher-jmeter/src/bom/build.gradle.kts' line: 90
    Cannot convert the provided notation to an object of type DependencyConstraint: DefaultExternalModuleDependency{group='org.codehaus.groovy', name='groovy-bom', version='3.0.3', configuration='default'}.
    The following types/formats are supported:
      - Instances of DependencyConstraint.
      - String or CharSequence values, for example 'org.gradle:gradle-core:1.0'.
      - Maps, for example [group: 'org.gradle', name: 'gradle-core', version: '1.0'].
      - Projects, for example project(':some:project:path').
      - Instances of ProjectDependency.
    
    Comprehensive documentation on dependency notations is available in DSL reference for DependencyHandler type.
        Caused by: org.gradle.internal.typeconversion.UnsupportedNotationException: Cannot convert the provided notation to an object of type DependencyConstraint: DefaultExternalModuleDependency{group='org.codehaus.groovy', name='groovy-bom', version='3.0.3', configuration='default'}.
        The following types/formats are supported:
          - Instances of DependencyConstraint.
          - String or CharSequence values, for example 'org.gradle:gradle-core:1.0'.
          - Maps, for example [group: 'org.gradle', name: 'gradle-core', version: '1.0'].
          - Projects, for example project(':some:project:path').
          - Instances of ProjectDependency.
```

---

Oh,

https://docs.gradle.org/current/userguide/java_platform_plugin.html#sec:java_platform_bom_import

So there should be the following before `dependencies`

```
javaPlatform {
    allowDependencies()
}
```

and `api(platform(...))` should come in the dependencies section rather than `dependencies { constraints { ... }}`.

Can you please try that? (move `api(platform` before `(constraints) {...`)

---

That works. Thanks for the hint.